js_modules/dagit/packages/core/src/nav/RepositoryLocationStateObserver.tsx

```diff
@@ -63,6 +67,19 @@ export const RepositoryLocationStateObserver = () => {
     return null;
   }
 
+  if (flagLiveReload) {
+    setUpdatedLocations([]);
+    refetch();
+    client.resetStore();
+    SharedToaster.show({
+      icon: 'done',
+      intent: Intent.SUCCESS,
+      message: `Repository location ${updatedLocations[0]} has been updated and automatically reloaded.`,
+      timeout: 3000,
+    });
+  }
```

Comment:
FWIW I do think there should be a browser setting because publishing updates to browser dagit and automatically refreshing browser dagit are separate things that users may want to independently vary. Right now browser dagit is already set up to capture published updates for any repo and alert the user to perform a manual reload. It's conceivable that they may want to keep this behavior but still receive published updates for local repos.

Reply:
Hmmmm, so I buy that you might want to configure how dagit handles what to do when one of the code servers updates - but I don't totally buy that the frontend should distinguish between the case where the server is externally managed or managed within Dagit. I think clearly differentiating between those two cases in a way that would be clear to the user sounds hard to explain.

Maybe worth a quick Slack huddle to talk it through?